src/openrct2/GameState.cpp

```diff
@@ -281,6 +281,7 @@ void GameState::UpdateLogic()
     scenario_update();
     climate_update();
     map_update_tiles();
+    map_calc_highest_tile_height(false);
```

Comment:
Each tick of the game loop we check a small set of map tiles for the tallest element.